### DISCUSSION TITLE
How to get release notes from CHANGELOG.md for a Python package?

### DISCUSSION THREAD

**Discussion by landier** [renovatebot/renovate#11668] [Request Help]
Hello,
I have a non-public Python package fetched from a private Python package index.
I'd like Renovate to display the changelog for that Python package on the PR it pushes to the different repositories that use that package.

I went across this [4724 issue](https://github.com/renovatebot/renovate/issues/4724) which mentions that the CHANGELOG.md must be included on the package. I guess this makes sense for a JS package but it does not for a Python package AFAIK.

Since the Python package source code is hosted on Github, how can I let Renovate know where it can find the `CHANGELOG.md` using Github?

**Comment by rarkins** [comment]
Best option is that you make some change to how you publish the package so that Renovate can fill the `sourceUrl` field automatically. If you search https://github.com/renovatebot/renovate/blob/main/lib/datasource/pypi/index.ts for `sourceUrl` you can see where the value comes from. If it points to github and Renovate has access to that repo too then you should see it find the CHANGELOG file.

Are you using the hosted app, or self hosting?

**Reply by landier** [reply]
OK very clear. I'll have a look at this `sourceUrl`.

We're using the hosted app.

Thank you.

**Reply by landier** [reply]
So I had a look and I'm not sure to understand what you mean by "make some change to how you publish the package".

Do you mean I should be referencing the package Github repo directly in my `requirements.txt` for the different projects so that Renovate find the `CHANGELOG.md` instead of using my private Python package repository?
I'm not sure to be able to do so.

Can't I specify somehow in `renovate.json` that my package source code is present at `https://github.com/abc/xyz`?

**Reply by rarkins** [reply]
What type of registry are you using?

**Reply by landier** [reply]
We're using https://gemfury.com/.

**Reply by rarkins** [reply]
So either:
- you are not providing the source url metadata to gemfury when publishing, so need to work out how to do that, or
- gemfury is using an API which doesn't allow inclusion of source url, in which case there's not much which can be done, or
- gemfury does not include the source url in its API metadata - in which case you need to file a feature request with them, or
- gemfury does include the source url in its API data but Renovate is failing to parse it. if you can use your token to manually get a copy of the data sent by gemfury then we can look into it

**Reply by landier** [reply]
Hello,
Thanks for your help.

Here is what the `PKG-INFO` looks like after I fetch it from gemfury so the source url seems presents as the `Home-page`:

```
Metadata-Version: 2.1
Name: company-python
Version: 0.32.0
Summary: Common resources for python projects.
Home-page: https://github.com/org/code-sdk
Author: Team
Author-email: team@email.com
License: UNKNOWN
Description: # python-sdk

        Python SDK is a set of libraries/utilies for Python developers.

        ## How to add `python-sdk` to your project
        Add these lines into your `requirements.txt`:

        ## TODO
        * Remove `http/monitoring/setup.py`
        * Create a folder `http/endpoints` and gather `/health` and `/metrics` inside

        # Changelog

Platform: any
```

**Reply by landier** [reply]
OK I got it, so it seems that Gemfury API does not provide the `/json` endpoint required that Renovate looks for.
